src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java

```diff
@@ -0,0 +1,442 @@
+    CdfGcsActions.skipHeader();
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with Path Field {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithPathField(String bucket, String format, String pathField)
+    throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterPathField(E2ETestUtils.pluginProp(pathField));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} " +
+    "with Override field {string} and data type {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithOverrideField
+    (String bucket, String format, String overrideField, String dataType) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterOverride(E2ETestUtils.pluginProp(overrideField));
+    CdfGcsActions.clickOverrideDataType(E2ETestUtils.pluginProp(dataType));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with delimiter field {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithDelimiterField(
+    String bucket, String format, String delimiter) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterDelimiterField(E2ETestUtils.pluginProp(delimiter));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} " +
+    "with minSplitSize {string} and maxSplitSize {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithMinSplitSizeAndMaxSplitSize(
+    String bucket, String format, String minSplitSize, String maxSplitSize) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterMaxSplitSize(E2ETestUtils.pluginProp(minSplitSize));
+    CdfGcsActions.enterMinSplitSize(E2ETestUtils.pluginProp(maxSplitSize));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with regex path filter {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithRegexPathFilter(
+    String bucket, String format, String regexPathFilter) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterRegexPath(E2ETestUtils.pluginProp(regexPathFilter));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with file encoding {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithFileEncoding(
+    String bucket, String format, String fileEncoding) throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.selectFileEncoding(fileEncoding);
+  }
+
+  @Then("Capture and validate output schema")
+  public void captureOutputSchema() {
```

Comment:
Use this generic method by passing the expected schema as param to validate the output schema, introduced in this PR. https://github.com/cdapio/cdap-e2e-tests/pull/52/files#diff-8077e12f3d9c0091d7faf79ff28b293df2ed40753e19da2497e9ae9c209140d3R178

Reply:
updated method